### DISCUSSION TITLE
Sketch on Face

### DISCUSSION THREAD

**Discussion by jgomez720** [KittyCAD/modeling-app#477] [New Feature Requests]
### Description

A tool that enables users to sketch on any flat face on a body. 

### Functionality

1. Either the user selects a face and then the sketch tool or a user selects the sketch tool and selects the face they want to sketch from. 

### Helpful Docs/Images
https://github.com/KittyCAD/modeling-app/assets/114548659/e5b7e66f-9d22-4601-abae-420191cb76d0

## PseudoKCL
```
const bracket = startSketchOn('XY')
  |> startProfileAt([0, 0])
  |> line([0, 5], %)
  |> line([8, 0], %, tag: 'face1')
  |> line([0, -0.25], %)
  |> line([-1 * 8.25, 0], %)
  |> line([0, -1 * 5.25], %)
  |> close(%)
  |> extrude(6, %)

const extrude = startSketchOn(bracket, 'face1')
  startProfileAt([0, 0])
  |> line([0, 3], %)
  |> line([3, 0], %)
  |> line([0, -3], %)
  |> close(%)
  |> extrude(3, %)
```

**further clarification**
https://github.com/KittyCAD/modeling-app/assets/114548659/2e6a6cde-5215-4c29-a797-4b93b8797438

**Comment by DeadBugEngineering** [comment]
Please allow users to choose between 'changing to normal view on the sketch plane' and 'leaving the view as it is' when starting a new sketch in the default settings of the app.

**Reply by jgomez720** [reply]
Good call @DeadBugEngineering. I believe this would be a front end thing for @franknoirot to throw in the settings.
Created this: https://github.com/KittyCAD/modeling-app/issues/1064

**Reply by franknoirot** [reply]
I can get this done pretty quick @jgomez720, sorry I must have missed your comment when I came back from vacation.

**Comment by keenanjohnson** [comment]
+1 for this idea. This is how both myself and most mechanical engineers are used to working.
